diffrax/_brownian/tree.py

```diff
@@ -206,14 +211,10 @@ def sqrt_mult(z):
             dtype = jnp.result_type(z)
             return jnp.astype(sqrt_len, dtype) * z
 
-        return LevyVal(
-            dt=jtu.tree_map(mult, x.dt),
-            W=jtu.tree_map(sqrt_mult, x.W),
-            H=jtu.tree_map(sqrt_mult, x.H),
-            bar_H=None,
-            K=jtu.tree_map(sqrt_mult, x.K),
-            bar_K=None,
-        )
+        leaves, treedef = jtu.tree_flatten(x)
+        dt_normalized = [mult(leaves[0])]
+        other_normalized = jtu.tree_map(sqrt_mult, leaves[1:])
+        return jtu.tree_unflatten(treedef, dt_normalized + other_normalized)
```

Comment:
So I think this is a bit dangerous, quietly assuming that `dt` is going to be the first return value. It's a little more effort, but you should be able to extract it like so:
```python
dt_normalised = mult(x.dt)
x = eqx.tree_at(lambda y: y.dt, x, None)
x = jtu.tree_map(sqrt_mult, x)
return eqx.tree_at(lambda y: y.dt, x, dt_normalised, is_leaf=lambda z: z is None)
```
which replaces the `dt` field with a `None` whilst applying the `sqrt_mult`, and then reinserts the normalised version back.

Reply:
I had originally attempted to use `eqx.partition` and `eqx.combine` with `is_leaf=lambda x: isinstance(x, AbstractLevyArea)`, and `filter_spec=lambda x: x.dt` but received TracerBooleanConversionErrors. Would you have expected this to fail? Also, would the tree_at approach actually occur in place, unlike the partition combine case?

Reply:
I think what's happening htere is that the `x.dt` in `filter_spec=lambda x: x.dt` is trying to be interpreted as a boolean. The intention of a filter spec is that it returns bool values to specify which half of the partition an element ends up in.
I think that could be made to work as well!

`tree_at` operates out of place. (Basically everything in JAX+Equinox does.)